application/src/main/kotlin/com/hanghe/redis/movie/MovieService.kt

```diff
@@ -1,32 +1,57 @@
 package com.hanghe.redis.movie
 
+import com.hanghe.redis.cache.CacheInfo
+import com.hanghe.redis.cache.CacheManager
 import com.hanghe.redis.movie.response.GetMovieScreeningResponses
 import com.hanghe.redis.mysql.movie.MovieRepository
 import com.hanghe.redis.mysql.screening.ScreeningRepository
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 
 @Service
-@Transactional
+@Transactional(readOnly = true)
 class MovieService(
     private val movieRepository: MovieRepository,
-    private val screeningRepository: ScreeningRepository
+    private val screeningRepository: ScreeningRepository,
+
+    private val cacheManager: CacheManager
 ) {
 
-    fun getAllScreeningMovies(): GetMovieScreeningResponses {
-        val allMovies = movieRepository.findAllByOrderByReleaseDateDesc()
+    fun getAllScreeningMovies(
+        title: String?,
+        genre: String?
+    ): GetMovieScreeningResponses {
+        validateParams(title, genre)
+
+        return if (title.isNullOrEmpty()) {
+            searchCached(title, genre)
+        } else {
+            search(title, genre)
+        }
```

Comment:
```suggestion
        return if (title.isNullOrEmpty()) {
            searchCached(title, genre)
        } else {
            search(title, genre)
        }
```

kotlin 스럽게 함수형으로 바꿔본다면 위처럼도 괜찮아보입니다.

Reply:
어떤 부분이 변경이 된것일까요..?! 😅

Reply:
@juhyun-leee 님, suggestion 에서 실수가있었네요ㅠ 죄송합니다. 아래처럼 함수형기반으로 바꿔봐도 좋겠다는 의견이었습니다.

```suggestion
        return 
        title?.takeIf { it.isNotBlank() }
        ?.let { searchCached(it, genre) }
        ?: search(title, genre)
```

Reply:
@ann-mj 아하~ 제안주신 코드로 변경하더라도 결과는 동일하게 동작하겠네요!

저도 함수형을 즐겨 사용하지만, 개인적으로는 기존 코드가 가독성적인 측면에서는 더 괜찮지 않을까 하는 개인적인 생각이 있어서요. 
요건 고려해보겠습니다 🙇